test/unit/spec/EnrollOnPrem_spec.js

```diff
@@ -60,7 +60,25 @@ function (Q, _, $, OktaAuth, Util, Form, Beacon, Expect, $sandbox,
       });
     }
 
-    var setupOnPrem = _.partial(setup, true);
+    var setupOnPrem = _.partial(setup, resAllFactorsOnPrem, true);
+
+    var getResponseNoProfile = function (response, factorType, provider) {
+      var copy = $.extend(true, {}, response);
```

Comment:
we should do a deepCopy similar to what we have in https://github.com/okta/okta-signin-widget/blob/master/test/unit/spec/EnrollChoices_spec.js#L56

nit: can we rename ```copy``` to ```res```  or ```responseCopy```